### PR TITLE
chore: send errors to Sentry

### PR DESCRIPTION
**Pull request by wa0x6e**
Setup sentry

- Send all unhandled errors to Sentry through Express error handler
- Send all captured errors via try/catch to Sentry

To test, put your own SENTRY_DSN in .env, and trigger an error by putting `throw new Error()` somewhere in the code

### CODE REVIEW

**Comment by Todmy** [comment]
I think you should move all Sentry initializations to a separate file. This will allow you to transfer the solution to other projects more conveniently. Another advantage of this is that third parties will be encapsulated, allowing us to modify it more easily.

File: src/index.ts [hunk 1]
```diff
@@ -14,6 +15,20 @@ import { rpcError } from './helpers/utils';
 const app = express();
 const PORT = process.env.PORT || 3005;
 
+Sentry.init({
+  dsn: process.env.SENTRY_DSN,
+  integrations: [
+    new Sentry.Integrations.Http({ tracing: true }),
+    new Sentry.Integrations.Express({ app }),
+    ...Sentry.autoDiscoverNodePerformanceMonitoringIntegrations()
```
Comment (Todmy) [inline src/index.ts]: it could send a lot of requests to the sentry (so we can exceed limits for the account) - we need to observe

**Comment by wa0x6e** [comment]
Should be better ?

**Review by Todmy — CHANGES_REQUESTED** [review]
I think it would be nice to configure [distributed tracing](https://blog.sentry.io/distributed-tracing-101-for-full-stack-developers/). It will allow us to track different request through all our apps

**Comment by Todmy** [comment]
> Should be better ?

much better 😊
but let's encapsulate it more:
```
import * as Sentry from '@sentry/node';
// ...
Sentry.captureException(e);
```
to
```
import logger from 'loger';
// ...
logger.capture(e);
```
I would also recommend renaming `sentry` to `logger` to make it more general

**Comment by wa0x6e** [comment]
I propose to keep the filename as "sentry", for easier searching. `logger` also does seem too generic, and may be confused with the simple logger like betterstack. We already use `msyql.ts`

**Comment by Todmy** [comment]
> I propose to keep the filename as "sentry", for easier searching. `logger` also does seem too generic, and may be confused with the simple logger like betterstack. We already use `msyql.ts`

but we are going to remove Betterstack, aren't we?

**Comment by wa0x6e** [comment]
> > I propose to keep the filename as "sentry", for easier searching. `logger` also does seem too generic, and may be confused with the simple logger like betterstack. We already use `msyql.ts`
> 
> but we are going to remove Betterstack, aren't we?

AFAIK, no, betterstack will still be used to track logs

**Comment by wa0x6e** [comment]
@Todmy any more changes needed for this PR ?

**Review by Todmy — APPROVED** [review]
tACK

**Pull request merged**